Any KCMinit module which are at X-KDE-Init-Phase=1 or 2 will now get
run. It also means kcminit_startup closes, instead of lingersing on
startup doing nothing until it times out 5 minutes later.

Test Plan: Confirmed kcminit_startup now exits properly.

Reviewers: #plasma, apol

Reviewed By: apol

Subscribers: plasma-devel

Tags: #plasma

Differential Revision: https://phabricator.kde.org/D2887
---


--- a/startkde/kcminit/main.cpp
+++ b/startkde/kcminit/main.cpp
@@ -212,6 +212,10 @@
 
      sendReady();
      QTimer::singleShot( 300 * 1000, qApp, &QCoreApplication::quit); // just in case
+
+     QDBusConnection::sessionBus().registerObject(QStringLiteral("/kcminit"), this, QDBusConnection::ExportScriptableContents);
+     QDBusConnection::sessionBus().registerService(QStringLiteral("org.kde.kcminit"));
+
      qApp->exec(); // wait for runPhase1() and runPhase2()
   }
   else
@@ -266,8 +270,6 @@
   about.processCommandLine(&parser);
 
   KCMInit kcminit( parser );
-  QDBusConnection::sessionBus().registerObject(QStringLiteral("/kcminit"), &kcminit, QDBusConnection::ExportScriptableContents);
-  QDBusConnection::sessionBus().registerService(QStringLiteral("org.kde.kcminit"));
   return 0;
 }
